include/secp256k1_silentpayments.h

```diff
@@ -0,0 +1,406 @@
+#ifndef SECP256K1_SILENTPAYMENTS_H
+#define SECP256K1_SILENTPAYMENTS_H
+
+#include "secp256k1.h"
+#include "secp256k1_extrakeys.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/* This module provides an implementation for Silent Payments, as specified in
+ * BIP352. This particularly involves the creation of input tweak data by
+ * summing up private or public keys and the derivation of a shared secret using
+ * Elliptic Curve Diffie-Hellman. Combined are either:
+ *   - spender's private keys and recipient's public key (a * B, sender side)
+ *   - spender's public keys and recipient's private key (A * b, recipient side)
+ * With this result, the necessary key material for ultimately creating/scanning
+ * or spending Silent Payment outputs can be determined.
+ *
+ * Note that this module is _not_ a full implementation of BIP352, as it
+ * inherently doesn't deal with higher-level concepts like addresses, output
+ * script types or transactions. The intent is to provide a module for
+ * abstracting away the elliptic-curve operations required for the protocol. For
+ * any wallet software already using libsecp256k1, this API should provide all
+ * the functions needed for a Silent Payments implementation without requiring
+ * any further elliptic-curve operations from the wallet.
+ */
+
+/* This struct serves as an In param for passing the silent payment address
```

Comment:
```suggestion
/* This struct serves as an input argument for passing the silent payment address
```

would be more consistent with secp256k1.h